src/main/scala/com/mozilla/telemetry/heka/package.scala

```diff
@@ -7,13 +7,95 @@
 package com.mozilla.telemetry
 
 import com.google.protobuf.ByteString
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.JsonDSL._
+import scala.util.{Try, Success, Failure}
 
 package object heka {
   class RichMessage(m: Message) {
+
+    /** Deserialize the Heka message as a native Scala Map. */
     def fieldsAsMap: Map[String, Any] = {
       val fields = m.fields
       Map(fields.map(_.name).zip(fields.map(field)): _*)
     }
+
+    /** Reconstructs a telemetry Heka message as a json4s JValue.
+      *
+      * The message is parsed and serialized into JSON. Keys that have been extracted
+      * into the toplevel document are inserted into to the proper place in the document.
+      *
+      * @return wrapped reconstructed JValue
+      */
+    def toJValue: Try[JValue] = {
```

Comment:
It's preferable to make a return value an Option, which is a top-level type in Scala -- there's a `.toOption` method on Try that might be useful for you